### PR TITLE
DDF-4589 Deprecate Metacard.* attributes

### PR DESCRIPTION
**Pull request by ethantmanns**
#### What does this PR do?
I still need to do some testing on this.

- Marks the attributes in the `Metacard` class as deprecated. The catalog taxonomy should be used as often as possible.
- Updates their documentation to point at the catalog taxonomy equivalent (if applicable)
- Replaces usages of the deprecated attribute constants with the catalog taxonomy equivalents
- Replaces usages the `getCreatedDate()` and `getModifiedDate()` and marks them as deprecated. This is due to the meaning of these attributes changing in the catalog taxonomy.

#### Who is reviewing it? 
@codice/data 

#### How should this be tested?
Full build

#### What are the relevant tickets?
For GH Issues:
Fixes: #4589 

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [X] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by ethantmanns** [comment]
build now

File: catalog/core/catalog-core-api-impl/src/test/java/ddf/catalog/data/impl/MetacardImplTest.java [hunk 1]
```diff
@@ -142,14 +143,14 @@ public void testMetacardImpl() {
     assertEquals(null, mi.getContentTypeName());
     assertEquals(null, mi.getContentTypeNamespace());
     assertEquals(null, mi.getContentTypeVersion());
-    assertEquals(null, mi.getCreatedDate());
+    assertEquals(null, mi.getAttribute(Core.CREATED).getValue());
```
Comment (bdeining) [inline catalog/core/catalog-core-api-impl/src/test/java/ddf/catalog/data/impl/MetacardImplTest.java]: why do these need to change?  the constant was updated https://github.com/codice/ddf/pull/4608/files#diff-c955ba16e1794205a33bc834339c7aa7R232

Comment (ethantmanns) [reply catalog/core/catalog-core-api-impl/src/test/java/ddf/catalog/data/impl/MetacardImplTest.java]: I wanted to remove the references to these methods because they are now ambiguous with the catalog taxonomy. The javadoc claims
>  Returns the date/time this {@link Metacard} was created.
However, the attribute that the method actually returns instead refers to the resource created date. This is due to the older taxonomy of "created" referring to metacard dates, while in the new taxonomy, "created" refers to the resource date, and instead "metacard.created" refers to the metacard date. 

Basically, I marked it as deprecated and removed its usage because it doesn't do what it was originally designed to do. We could update it to instead return the new "metacard.modified" attribute, but I'm not sure what downstream behavior is expecting at this point. Every usage out there is either ignoring the javadoc, or is not doing what was intended.

**Comment by Lambeaux** [comment]
❓ Downstream aside, is there anywhere in DDF we're relying on the incorrect behavior of the metacard methods? Were those cases ignored?

**Comment by ethantmanns** [comment]
@Lambeaux All the uses of those methods were updated to not use the methods, but still rely on the same attribute for the date. So if they're relying on the behavior being incorrect (from the method's perspective), their behavior will remain the same. Instead they're directly pulling the resource dates.

File: catalog/confluence/catalog-confluence-source/src/test/java/org/codice/ddf/confluence/source/ConfluenceInputTransformerTest.java [hunk 2]
```diff
@@ -190,8 +192,10 @@ private void validatePageMetacard(
     assertThat(
         mcard.getAttribute(Associations.EXTERNAL).getValues(),
         contains("https://codice.atlassian.net/wiki/display/DDF/Formatting+Source+Code"));
-    assertThat(mcard.getCreatedDate(), is(getDate("2013-09-18T14:50:42.616-07:00")));
-    assertThat(mcard.getModifiedDate(), is(getDate("2015-06-16T19:21:39.141-07:00")));
+    assertThat(
+        mcard.getAttribute(Core.CREATED).getValue(), is(getDate("2013-09-18T14:50:42.616-07:00")));
+    assertThat(
+        mcard.getAttribute(Core.MODIFIED).getValue(), is(getDate("2015-06-16T19:21:39.141-07:00")));
```
Comment (Lambeaux) [inline catalog/confluence/catalog-confluence-source/src/test/java/org/codice/ddf/confluence/source/ConfluenceInputTransformerTest.java]: ❓@ethantmanns I'm a tad confused why the above methods are deprecated if, according to these test assertions, they are equivalent to directly going through the taxonomy.

Comment (Lambeaux) [reply catalog/confluence/catalog-confluence-source/src/test/java/org/codice/ddf/confluence/source/ConfluenceInputTransformerTest.java]: ❓Also I'm a tad confused why no additional code needs to be introduced if code depends on the deprecated semantics of the metacard date methods.

Comment (ethantmanns) [reply catalog/confluence/catalog-confluence-source/src/test/java/org/codice/ddf/confluence/source/ConfluenceInputTransformerTest.java]: The API documentation is wrong because those taxonomy terms no longer represent what the method claims to return

Comment (ethantmanns) [reply catalog/confluence/catalog-confluence-source/src/test/java/org/codice/ddf/confluence/source/ConfluenceInputTransformerTest.java]: It claims to return the metacard date, but it returns the resource date. This is because that same attribute name previously represented the metacard date, but now represents the resource date.
